ports/zephyr/main.c

```diff
@@ -58,6 +58,19 @@
 #include "extmod/vfs.h"
 #endif
 
+#if MICROPY_ENABLE_GC
+#if defined(MICROPY_GC_SPLIT_HEAP) && defined(CONFIG_MICROPY_ADDITIONAL_HEAPS)
```

Comment:
Maybe just use `#if MICROPY_GC_SPLIT_HEAP`?  Because if you enable split heap you surely wanted a split heap and a build error would be helpful if the list of heaps is not defined.

(Also using `defined(MICROPY_GC_SPLIT_HEAP)` is not correct because the macro is always defined, just not always defined to 1.)